plugins/utils/which-key.nix

```diff
@@ -559,26 +559,72 @@ lib.nixvim.neovim-plugin.mkNeovimPlugin config {
       '';
       visible = false;
     };
+
+    mappingsWithOpts = mkOption {
+      type =
+        with types;
+        listOf (submodule {
+          options = {
+            spec = mkOption {
+              type = listOf (attrsOf anything);
+              description = ''
+                WhichKey mappings following the same setup as `plugins.which-key.settings.spec`.
+              '';
+            };
+            opts = mkOption {
+              type = attrsOf anything;
+              description = ''
+                WhichKey mapping options to apply to all the mapping in corresponding `plugins.which-key.mappingsWithOpts`.
+              '';
+            };
+          };
+        });
+      description = ''
+        Allows setting registrations with shared opts.
+      '';
+      example = [
+        {
+          spec = [
+            {
+              __unkeyed-1 = "<leader>f";
+              group = "Search";
+            }
+            {
+              __unkeyed-1 = "<leader>fb";
+              group = "Bindings";
+            }
+          ];
+          opts = {
+            mode = "v";
+          };
+        }
+      ];
+    };
   };
 
   # TODO: introduced 2024-07-29: remove after 24.11
   # NOTE: this may be upgraded to a mkRemoveOptionModule when which-key removes support
-  extraConfig =
-    cfg:
-    lib.mkIf opt.registrations.isDefined {
-      warnings = [
-        ''
-          nixvim (plugins.which-key):
-          The option definition `plugins.which-key.registrations' in ${showFiles opt.registrations.files} has been deprecated in which-key v3; please remove it.
-          You should use `plugins.which-key.settings.spec' instead.
-
-          Note: the spec format has changed in which-key v3
-          See: https://github.com/folke/which-key.nvim?tab=readme-ov-file#%EF%B8%8F-mappings
-        ''
-      ];
+  extraConfig = cfg: {
+    warnings = lib.mkIf opt.registrations.isDefined [
+      ''
+        nixvim (plugins.which-key):
+        The option definition `plugins.which-key.registrations' in ${showFiles opt.registrations.files} has been deprecated in which-key v3; please remove it.
+        You should use `plugins.which-key.settings.spec' instead.
+
+        Note: the spec format has changed in which-key v3
+        See: https://github.com/folke/which-key.nvim?tab=readme-ov-file#%EF%B8%8F-mappings
+      ''
+    ];
 
-      extraConfigLua = lib.optionalString opt.registrations.isDefined ''
-        require("which-key").register(${toLuaObject cfg.registrations})
-      '';
-    };
+    extraConfigLua =
+      lib.optionalString opt.registrations.isDefined ''
+        require("which-key").register(${builtins.toJSON cfg.registrations})
+      ''
+      + lib.optionalString opt.mappingsWithOpts.isDefined (
+        lib.concatMapStringsSep "\n" (
+          regWithOpts:
+          "require('which-key').add(${toLuaObject regWithOpts.spec},${toLuaObject regWithOpts.opts})"
+        ) cfg.mappingsWithOpts
```

Comment:
This should work fine, but _usually_ we assign the whole option to a local variable, and then loop over it using `for in pairs`.

E.g. https://github.com/nix-community/nixvim/blob/a96aa9730af8c85dd7ed15e359ac23e9686f0a9a/modules/opts.nix#L74-L82

Or: https://github.com/nix-community/nixvim/blob/a96aa9730af8c85dd7ed15e359ac23e9686f0a9a/plugins/utils/wilder.nix#L257-L263

Reply:
I can follow the pattern. I have an update using this.